### PR TITLE
[MerkleDB] Make intermediate node cache two layered

### PR DESCRIPTION
**Pull request by dboehm-avalabs**
The flushing of the intermediate nodes on db close is taking too long.  Split the cache into two parts, one normal cache that can be large and store nodes in memory and one smaller delayed write cache that will prevent writes, but wont be too large to flush on database close.

### CODE REVIEW

File: x/merkledb/intermediate_node_db.go [hunk 1]
```diff
@@ -27,7 +29,9 @@ type intermediateNodeDB struct {
 	// from the cache, which will call [OnEviction].
 	// A non-nil error returned from Put is considered fatal.
 	// Keys in [nodeCache] aren't prefixed with [intermediateNodePrefix].
-	nodeCache onEvictCache[Key, *node]
+	writeBuffer onEvictCache[Key, *node]
+
+	nodeCache cache.Cacher[Key, *node]
```
Comment (patrick-ogrady) [inline x/merkledb/intermediate_node_db.go:34]: If the comment above `writeBuffer` also applies to `nodeCache` I think you should remove the `\n` above.

Comment (dboehm-avalabs) [reply x/merkledb/intermediate_node_db.go:34]: nodecache wont cause writes on evictions, but the thing about nils is still true, may just replicate the parts that apply.

File: x/merkledb/db.go [hunk 2]
```diff
@@ -483,19 +486,11 @@ func (db *merkleDB) PrefetchPath(key []byte) error {
 
 func (db *merkleDB) prefetchPath(keyBytes []byte) error {
 	return visitPathToKey(db, ToKey(keyBytes), func(n *node) error {
-		if !n.hasValue() {
-			// this value is already in the cache, so skip writing
```
Comment (dboehm-avalabs) [inline x/merkledb/db.go:487]: the lru cache does lock on the Get call, so there is no benefit of checking presence first.

Comment (patrick-ogrady) [reply x/merkledb/db.go:487]: The benefit is that the Get call only uses an RLock, so this can be much more concurrent friendly.

Comment (dboehm-avalabs) [reply x/merkledb/db.go:487]: https://github.com/ava-labs/avalanchego/blob/master/cache/lru_sized_cache.go#L41

File: x/merkledb/proof.go [hunk 3]
```diff
@@ -861,11 +861,11 @@ func getStandaloneView(ctx context.Context, ops []database.BatchOp, size int) (*
 		ctx,
 		memdb.New(),
 		Config{
-			EvictionBatchSize:         verificationEvictionBatchSize,
-			Tracer:                    trace.Noop,
-			ValueNodeCacheSize:        verificationCacheSize,
-			IntermediateNodeCacheSize: verificationCacheSize,
-			BranchFactor:              tokenSizeToBranchFactor[size],
+			IntermediateWriteBatchSize: verificationIntermediateWriteBatchSize,
```
Comment (danlaine) [inline x/merkledb/proof.go:864]: Should we set `IntermediateWriteBufferSize` here too?

**Review by danlaine — APPROVED** [review]

**Pull request merged**